lib/fog/vsphere/models/compute/snapshot.rb

```diff
@@ -0,0 +1,62 @@
+require 'fog/compute/models/server'
+
+module Fog
+  module Compute
+    class Vsphere
+      class Snapshot < Fog::Model
+
+        identity  :ref
+        attribute :server_id
+
+        attribute :name
+        attribute :quiesced
+        attribute :description
+        attribute :create_time
+        attribute :power_state
+        attribute :ref
+        attribute :mo_ref
+        attribute :tree_node
+        attribute :snapshot_name_chain
+        attribute :ref_chain
+
+        attr_reader :service
+
+        def initialize(attributes = {})
+          super defaults.merge(attributes)
+          requires :mo_ref
```

Comment:
The i would say that this `requires` should be inside `child_snapshots` and not in the constructor.